new_client/src/components/events/Event.jsx

```diff
@@ -0,0 +1,74 @@
+import axios from 'axios';
+import React, { Component } from 'react';
+import AddEvent from './AddEvent';
+import EventList from './EventList';
+
+class Event extends Component {
+	state = {
+		events: [],
+		isUpdating: false,
+		updatingEvent: null,
+	};
+
+	handleEdit = (eventId) => {
+		const updatingEvent = this.state.events.find(
+			(event) => event._id === eventId
+		);
+		this.setState({ isUpdating: true, updatingEvent: updatingEvent });
+	};
+
+	handleDelete = (eventId) => {
+		axios.delete(process.env.REACT_APP_API + `/events/${eventId}`);
+		const deletedEvent = this.state.events.find(
+			(event) => event._id === eventId
+		);
+		if (
+			JSON.stringify(deletedEvent) ===
+			JSON.stringify(this.state.updatingEvent)
+		) {
+			this.setState({ isUpdating: false, updatingEvent: null });
+		}
+		this.setState({ events: [] });
+	};
+
+	componentDidMount() {
+		axios
+			.get(process.env.REACT_APP_API + '/events')
+			.then((res) => {
+				this.setState({ events: res.data });
+			})
+			.catch((error) => console.log(error));
+	}
+
+	componentDidUpdate(prevProps, prevState) {
+		if (
+			JSON.stringify(this.state.events) !==
+			JSON.stringify(prevState.events)
+		) {
+			axios
+				.get(process.env.REACT_APP_API + '/events')
+				.then((res) => {
+					this.setState({ events: res.data });
+				})
+				.catch((error) => console.log(error));
+		}
+	}
+

```

Comment:
Duplication could have been avoided if you had used hooks.

Reply:
Yes, I realized it later after more testing that there is no need for that piece of code so I've removed it.